torch/ao/quantization/fx/quantization_patterns.py

```diff
@@ -51,16 +51,22 @@ def __init__(
         """ Records pattern information in __init__, which will be used
         in convert
         """
-        # this is an indicator of whether all the inputs are Node or not
-        # since some op might be quantized differently depending on whether
-        # all inputs are tensors or not, e.g. add/mul
+        self.node_pattern = node_pattern
+        self.modules = modules
         if root_node_getter is None:
             root_node_getter = _default_root_node_getter
         self.root_node = root_node_getter(node_pattern)
+        self.num_tensor_args = 0
+        # determine how many of the first two args are Tensors (versus scalars)
+        # this distinguishes things like "x + y" from "x + 2" or "2 + x"
         if isinstance(self.root_node, Node):
-            self.num_tensor_args = len(self.root_node.args)
-        else:
-            self.num_tensor_args = 0
+            cache_for_no_tensor_check: Dict[Node, bool] = dict()
+            for arg_idx in range(len(self.root_node.args)):
+                arg = self.root_node.args[arg_idx]
+                if isinstance(arg, Node) and (
+                        not all_node_args_have_no_tensors(
+                            arg, self.modules, cache_for_no_tensor_check)):
```

Comment:
nit: This logic looks quite complicated. Maybe in a future PR we can rewrite `all_node_args_have_no_tensors` to instead return the number of tensor args instead, so we don't need to call this recursive function in a loop. I'm also not sure if we need the `cache` argument, or at least we should avoid exposing it to the caller.

Reply:
yeah both suggestions sounds good